### ISSUE TITLE
Not properly reacting to clicks of top left corner pixel

### ISSUE THREAD

**Issue by anohren** [elementary/applications-menu#390]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug

The topmost leftmost pixel does not consistently react to clicks. Just like every other gnome based desktop environment, there is about a 10% chance of no effect upon clicking. Not ensuring 100% reliable interaction with the most important pixel on the screen undermines the confidence in the design effort of the rest of the system. 

I did not find any alternative feedback channels so I'm reporting this as a bug, because I can't stand it. It can singlehandedly make me not use a gnome OS.

### To Reproduce

1. Move cursor to top left corner
2. Click on topmost leftmost pixel
3. If the menu reacts, move cursor up and left and click again
4. Repeat until eventually you'll get into a state where no clicks register
5. Experience significantly reduced confidence in one of the most fundamental ui elements in the system

### Expected behavior

Consistent reaction to clicks on corner pixel 100% of the time. 

## Platform Information

 - OS: elementary OS
 - OS Version: 5.1.4
 - Hardware info: nvidia GTX 550 ti (390 driver), athlon 3700+ (missing CMPXCHG16B), logitech mx 518 usb mouse

Please check what applies:

- [ ] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

## Additional context

It is widespread in every gnome distro, but for some reason I half expected it to be resolved here

**Comment by cassidyjames** [comment]
>Just like every other gnome based desktop environment, there is about a 10% chance of no effect upon clicking… It can singlehandedly make me not use a gnome OS.

elementary OS does not use the GNOME desktop or GNOME Shell, so I'm not sure how that's relevant.

Regardless, I can't reproduce this. Could you please share some more system information, including screen resolution/size and if you're using any third-party panel indicators, stylesheet, etc.?

**Comment by anohren** [comment]
Ok, it looked and behaved similar to the Activities menu in Ubuntu and the like so I thought they were related. Feel free to disregard that if there is no relation, although it might just point at some other shared component.

It's a freshly installed system, and the only thing I've installed is `sudo apt install software-properties-gtk software-properties-common` in order to select the correct nvidia driver after installation. I've not touched/customized anything else.

Screen resolution is 1680x1050.

**Comment by anohren** [comment]
Apparently I had typed hardware info inside a comment in the issue template. I moved it out in the original post.

**Comment by anohren** [comment]
![EB6B7360-85D1-472C-AD75-5EEC70757911](https://user-images.githubusercontent.com/3519879/83053072-72ad2880-a050-11ea-9e3f-709caeaa46e6.jpeg)

**Comment by anohren** [comment]
[IMG_8178.MOV.zip](https://github.com/elementary/applications-menu/files/4690719/IMG_8178.MOV.zip)

**Comment by anohren** [comment]
I have a video of the issue if it would be of any use.

**Comment by cassidyjames** [comment]
@anohren sure! You'll need to upload it somewhere like YouTube or Vimeo and share the link here.

**Comment by anohren** [comment]
[IMG_8178 2.MOV.zip](https://github.com/elementary/applications-menu/files/4692639/IMG_8178.2.MOV.zip)

Also some additional info: the problem occurrs for any edge pixel, left or top, not just the corner pixel.

**Comment by cassidyjames** [comment]
@anohren do you have multiple displays? Have you modified the panel or OS in any way?

**Comment by anohren** [comment]
No customization or modification. One single display.

**Comment by anohren** [comment]
As I mentioned, deceptively similar symptoms occur on other distros as well. Not only that, these edge interaction problems occurs on other machines as well, with or without multi monitor setups. 

For example, in e.g. Ubuntu and Ubuntu Mate running on a netbook or a macbook, Plank dock cannot properly unhide from its autohidden state when applying edge pressure with the cursor without the dock starting to twitch madly back and forth, and then either ending up hidden again, or visible, much like how the problem described in this issue can end up in one of two states. If I remember correctly, the Plank dock glitches always appear when using touchpads, and most often when moving the cursor at a certain speed. The problem in this issue occurs easily even with a normal usb mouse, which makes it doubly frustrating. Not sure if there are any clues in there for you to intuit your way to the root cause, but I can't rule out that they're related somehow. 

In summary it seems that most distros fail to properly decide whether the cursor actually exists on top of an edge pixel at all. That has been my impression for several years, and elementary OS is just the latest to display the same symptoms. 

Let me know if you need more details.

**Comment by danrabbit** [comment]
FWIW, I’m also unable to reproduce this issue

**Comment by anohren** [comment]
I'm afraid I cannot help you reproduce this without any details of what you've tried. With more info about that I could attempt to identify further differences that might be relevant. Unless you aldready know what information is missing of course, in which case you need only ask.

**Comment by jlnr** [comment]
I have encountered this issue a few times in the OS 5 days, and I just ran into it again. Fresh installation of the 2023-08-29 Daily ISO, and in some pixel positions, I was unable to click to open the Applications menu. Moving a bit to the left or right fixed it.

This is with Nvidia drivers, if it matters.

**Comment by jlnr** [comment]
Ah, it also happens with some of the indicator icons, so I guess I'll bury this old issue again. I will report back in https://github.com/elementary/wingpanel/issues/41 if I can figure out what is happening.

**Comment by anohren** [comment]
> Ah, it also happens with some of the indicator icons, so I guess I'll bury this old issue again. I will report back in [elementary/wingpanel#41](https://github.com/elementary/wingpanel/issues/41), which seems more appropriate, if I can figure out what is happening.

To be honest, I've had this happen on practically every single distribution using X (never on Wayland) on every machine and every kind of graphics hardware (well, both Intel and nvidia, as far as I can tell) for almost a decade. As such it's hard to think that this isn't an easily reproducible and widespread problem. 

Everything about edge interaction in X has been bad: ignored edge and corner pixel clicks, and an auto hiding dock twitching back and forth from the edge while slowly applying edge pressure with the cursor. 

It's as if the collision detection is rounding up or down to the nearest integer at random, while leaving the visible cursor in place. 

I've stopped using X.

**Comment by jlnr** [comment]
Ah, good point. It's indeed weird that the cursor doesn't visibly move between y=0 and y=-0.98 (see https://github.com/elementary/wingpanel/issues/41 - GTK is giving me fractional values).

Has this also happened with KDE? I'm still thinking it might be a GTK thing since X11 has integer mouse coordinates. But even then, it's good to know that it doesn't happen on Wayland.

**Comment by anohren** [comment]
Interesting. I've only been using gnome while experiencing it I think, but due to my impression of the nature of the bug I would've been surprised if it didn't show up in KDE.  

I commend your efforts in elimination and willingness to zero in on likely causes. I've not seen that in other places that I've reported this...phenomenon (bug-cloud?)

To expand on my impression, it behaves as though the cursor coordinates somehow exist in an unbounded space (even though the visible cursor stops at screen edges) and that it's just as likely for it to end on a decimal component that's below or above 0.5, and then naively rounding to the nearest integer, yielding a 50% chance to not hit the actual edge pixel. 

If memory serves the problem could sometimes be exacerbated or made to go away depending on single/multi-monitor, as well as multi desktop topology. E.g. when a dock was on a left edge and the two screens did not have aligned left edges, or something of that kind. I don't think I ever found an obvious pattern though.

**Comment by anohren** [comment]
I just took a look in the linked issue and see that you were on the same lines as me regarding erroneously rounding coordinates off-screen. 

From my perspective, having done no measurements other than having experienced it for years and having experimented with it (I seem to remember taking several videos clearly demonstrating the problem), I think it's a good working hypothesis that fits the facts.
